Emma/emma_libs/memoryManager.py

```diff
@@ -321,17 +340,25 @@ def _createTeamScalePath(memEntryRow):
             # TODO: Implement handling and choosing of which reports to create (via cmd line argument (like a comma separated string) (MSc)
             createStandardReports()
             createTeamScaleReports()
-
-            if image:
-                if noprompt:
-                    sc().wwarning("no prompt is active. No svg report will be created")
-                else:
-                    startRegion = input("Enter the start address of the region to be plotted (start with `0x` for hex; otherwise dec is assumed) \n")
-                    if "0x" in startRegion:
+            svgReport = False
+            if memVis or memVisResolved:
+                svgReport = True
+            if svgReport and noprompt:
+                sc().wwarning("No prompt is active. No SVG report will be created")
+            elif svgReport and noprompt is False:
+                while True:
+                    print("Enter the start address of the region to be plotted (start with `0x` for hex; otherwise dec is assumed):")
+                    startRegion = input("> ")
+                    if startRegion.startswith("0x"):
                         startRegion = int(startRegion, 16)
-                    endRegion = input("Enter the end address of the region to be plotted (start with `0x` for hex; otherwise dec is assumed) \n")
-                    if "0x" in endRegion:
+                    print("Enter the end address of the region to be plotted (start with `0x` for hex; otherwise dec is assumed):")
+                    endRegion = input("> ")
+                    if endRegion.startswith("0x"):
                         endRegion = int(endRegion, 16)
-                    createSvgReport(int(startRegion), int(endRegion))
+                    if str(startRegion).isdigit() and str(endRegion).isdigit():
+                        break
+                    else:
+                        sc().wwarning("The input is not valid number. Please enter the start and end address again \n")
+                createSvgReport(int(startRegion), int(endRegion))
```

Comment:
Conversion can be done here:

https://github.com/bmwcarit/Emma/blob/5c76a1392442adab1ee31832195b6da290c0f3f9/Emma/emma_libs/memoryManager.py#L368

Reply:
Now we convert two times. This is superfluous.